dataland-frontend/src/components/forms/CreateAdditionalCompanyInformationDataset.vue

```diff
@@ -0,0 +1,369 @@
+      const additionalCompanyInformationDataControllerApi = this.buildAdditionalCompanyInformationDataApi();
+      if (additionalCompanyInformationDataControllerApi) {
+        const dataResponse = await additionalCompanyInformationDataControllerApi.getFrameworkData(dataId);
+        const additionalCompanyInformationResponseData = dataResponse.data;
+        this.listOfFilledKpis = getFilledKpis(additionalCompanyInformationResponseData.data);
+        this.referencedReportsForPrefill =
+          additionalCompanyInformationResponseData.data?.general?.general?.referencedReports ?? {};
+        this.companyAssociatedAdditionalCompanyInformationData = objectDropNull(
+          additionalCompanyInformationResponseData
+        ) as CompanyAssociatedDataAdditionalCompanyInformationData;
+        this.waitingForData = false;
+      }
+    },
+    /**
+     * Sends data to add AdditionalCompanyInformation data
+     */
+    async postAdditionalCompanyInformationData(): Promise<void> {
+      this.messageCounter++;
+      try {
+        if (this.fieldSpecificDocuments.get(referenceableReportsFieldId)?.length) {
+          checkIfAllUploadedReportsAreReferencedInDataModel(
+            this.companyAssociatedAdditionalCompanyInformationData.data as ObjectType,
+            this.namesOfAllCompanyReportsForTheDataset
+          );
+        }
+        const documentsToUpload = Array.from(this.fieldSpecificDocuments.values()).flat();
+        await uploadFiles(documentsToUpload, assertDefined(this.getKeycloakPromise));
+
+        const additionalCompanyInformationDataControllerApi = this.buildAdditionalCompanyInformationDataApi();
+
+        const isCompanyOwnerOrDataUploader = await hasUserCompanyOwnerOrDataUploaderRole(
+          this.companyAssociatedAdditionalCompanyInformationData.companyId,
+          this.getKeycloakPromise
+        );
+
+        await additionalCompanyInformationDataControllerApi?.postFrameworkData(
+          this.companyAssociatedAdditionalCompanyInformationData,
+          isCompanyOwnerOrDataUploader
+        );
+
+        this.$emit('datasetCreated');
+        this.message = 'Upload successfully executed.';
+        this.uploadSucceded = true;
+      } catch (error) {
+        console.error(error);
+        if ((error as Error).message) {
+          this.message = formatAxiosErrorMessage(error as Error);
+        } else {
+          this.message =
+            'An unexpected error occurred. Please try again or contact the support team if the issue persists.';
+        }
+        this.uploadSucceded = false;
+      } finally {
+        this.postAdditionalCompanyInformationDataProcessed = true;
+      }
+    },
+    /**
+     * updates the list of documents that were uploaded
+     * @param reportsNamesAndReferences reports names and references
+     * @param reportsToUpload reports to upload
+     */
+    updateDocumentsList(reportsNamesAndReferences: object, reportsToUpload: DocumentToUpload[]) {
```

Comment:
Die Funktionen gibt es identisch bei SFDR. Die sollten ausgelagert werden um Codeduplikate zu vermeiden.

Reply:
Ich bin mir nicht sicher, wie das geht. Es ergibt aus meiner Sicht nicht so viel Sinn, nur die Logik auszulagern, sondern wenn ich dich richtig versteh, sollten wir die gesamten Funktionen auslagern. Die Funktionen werden aber von Komponenten-spezifischen Events getriggert, und innerhalb der Funktionen werden Komponenten-spezifische Felder gesetzt. Wie kann ich aus einer externen Utils-Klasse diese Felder der Komponente setzen?

Reply:
So, ich hab jetzt nochmal recherchiert. Generell ist der Reuse von Code möglich mithilfe von sogenannte Composables, die auch verwendbar sind, wenn man wie wir den Options-API-Flavor von Vue nutzt; siehe [https://vuejs.org/guide/reusability/composables.html#using-composables-in-options-api](https://vuejs.org/guide/reusability/composables.html#using-composables-in-options-api)

Dafür muss man die Felder in der setup setzen und dann returnen, damit sie an `this` angehängt werden, bspw: 

```
setup() {
    const namesAndReferencesOfAllCompanyReportsForTheDataset = useFirstExternUpdateFunction()
    const fieldSpecificDocuments = useSecondExternUpdateFunction()
   return {
      getKeycloakPromise: inject<() => Promise<Keycloak>>('getKeycloakPromise'),
      namesAndReferencesOfAllCompanyReportsForTheDataset,
      fieldSpecificDocuments,
    };
```

Gleichzeitig erforderte das im konkreten Fall ein etwas größeres Refactoring, denn das Feld `fieldSpecificDocuments` wird aktuell von zwei Funktionen gesetzt: der `updateDocumentsList` und der `updateDocumentsOnField`. Auf der anderen Seite setzt `updateDocumentsList` auch beide Felder. Das müsste man imo trennen/vereinheitlichen, damit das sauber läuft. Mein Vorschlag wäre jetzt, dass wir hierfür ein Technical-Debt-Ticket aufmachen und das nicht innerhalb dieses Tickets lösen.

Reply:
können wir so machen

Reply:
https://jira.d-fine.dev/projects/DALA/issues/DALA-4794

Reply:
Ich schließe hier, da das entpsrechende Ticket erstellt wurde.